src/treeview/tree-node.component.ts

```diff
@@ -207,8 +207,10 @@ export class TreeNodeComponent implements AfterContentChecked, OnInit, OnDestroy
 			this.selected = true;
 			this.active = true;
 			event.target.parentElement.focus();
+			const node = { id: this.id, label: this.label, value: this.value }
 			// Passes event to all nodes to update highlighting & parent to emit
-			this.treeViewService.selectNode({ id: this.id, label: this.label, value: this.value });
+			this.treeViewService.selectNode(node);
+			this.nodeSelect.emit({node, event})
```

Comment:
```suggestion
			this.nodeSelect.emit({...node, event});
```
Could you use the spread operator on node? This is a breaking change for existing users.

Reply:
Would need to update the interface as well

Reply:
at the moment the interface is returning an object with a property `node` and another `event`...if i use the spread operator there i would return a node with inside an event, but this would be a breaking change with the current interface

Reply:
oh sorry...now i see the breaking change for the select...yes, i'll push just the node without the event